src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryOfTKeyTValueConverter.cs

```diff
@@ -61,7 +61,7 @@ protected override void CreateCollection(ref Utf8JsonReader reader, ref ReadStac
                 do
                 {
                     TKey key = enumerator.Current.Key;
-                    _keyConverter.WriteWithQuotes(writer, key, options, ref state);
+                    _keyConverter.WriteToPropertyName(writer, key, options, ref state);
```

Comment:
Wanna update the `Read/Write...` methods in accordance with API review in this PR?

Reply:
Yep done.